### PR TITLE
RATEST-200 E2E Location management workflow test

### PR DESCRIPTION
**Pull request by Ndacyayisenga-droid**
Ticked id **RATEST-200**
Ticket url  https://issues.openmrs.org/browse/RATEST-200

### CODE REVIEW

File: .github/workflows/refapp-2x-location-management.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        with:
+          repository: ${{github.repository}}
+      - name: run db and web containers
+        run: |
+          docker-compose -f docker/docker-compose-refqa.yml up -d
+      - name: wait for openmrs instance to start
+        run: while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' http://localhost:8080/openmrs/login.htm)" != "200" ]]; do sleep 1; done
+      - name: Run qaframework on selenium
+        run: |
+          echo "cucumber.publish.enabled=true" > src/test/resources/cucumber.properties
```
Comment (kdaud) [inline .github/workflows/refapp-2x-location-management.yml:35]:
```suggestion
          echo "cucumber.publish.enabled=true" > qaframework-bdd-tests/src/test/resources/cucumber.properties
```

File: .github/workflows/refapp-2x-location-management.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          mvn clean install -DskipTests=true
+          cp -f src/test/resources/org/openmrs/uitestframework/test-local.properties src/test/resources/org/openmrs/uitestframework/test.properties
```
Comment (kdaud) [inline .github/workflows/refapp-2x-location-management.yml:37]:
```suggestion
          cp -f qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test-local.properties qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test.properties
```

**Comment by Ndacyayisenga-droid** [comment]
Hey @kdaud @sherrif10 Am failing to get [this](https://github.com/openmrs/openmrs-contrib-qaframework/blob/7a792d7fbf71a02cead899143e22d7e144d10810/qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java#L85) method working... 
I need help to get [this](https://github.com/openmrs/openmrs-contrib-qaframework/blob/7a792d7fbf71a02cead899143e22d7e144d10810/qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java#L85) working..... I don't know whether its a bug in the reference application or the [method](https://github.com/openmrs/openmrs-contrib-qaframework/blob/67663b68d6fd5b2770aea1cab57b33bd568e6fc4/qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/ManageLocationAttributeTypesPage.java#L72) from pages itself has issues. Am somewhat blocked. Thankyou

**Comment by mherman22** [comment]
> Hey @kdaud @sherrif10 Am failing to get [this](https://github.com/openmrs/openmrs-contrib-qaframework/blob/7a792d7fbf71a02cead899143e22d7e144d10810/qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java#L85) method working... I need help to get [this](https://github.com/openmrs/openmrs-contrib-qaframework/blob/7a792d7fbf71a02cead899143e22d7e144d10810/qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java#L85) working..... I don't know whether its a bug in the reference application or the [method](https://github.com/openmrs/openmrs-contrib-qaframework/blob/67663b68d6fd5b2770aea1cab57b33bd568e6fc4/qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/ManageLocationAttributeTypesPage.java#L72) from pages itself has issues. Am somewhat blocked. This [step](https://github.com/openmrs/openmrs-contrib-qaframework/blob/7a792d7fbf71a02cead899143e22d7e144d10810/qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java#L85) doesnt pass both locally and on github Thankyou. cc @mherman22 @Muta-Jonathan

Share the logs

**Comment by sherrif10** [comment]
can you share your logs after spinning this workflow locally

**Comment by Ndacyayisenga-droid** [comment]
> can you share your logs after spinning this workflow locally

@mherman22 @sherrif10 @kdaud  these are the [logs](https://pastebin.com/qwPkavML)

**Comment by Muta-Jonathan** [comment]
@Ndacyayisenga-droid due to 
>LOCATION_MANAGEMENT(Location Management)  Time elapsed: 11.822 sec  <<< ERROR!
org.openqa.selenium.support.ui.UnexpectedTagNameException: Element should have been "select" but was "textarea"

the test fails 

This error message...
>org.openqa.selenium.support.ui.UnexpectedTagNameException: Element should have been "select" but was "textarea"

implies that you have used `Select` class to interact with the desired element whereas the element was a ` textarea`.

try to use `WebDriverWait()` or use [this ](https://stackoverflow.com/questions/56471007/org-openqa-selenium-support-ui-unexpectedtagnameexception-element-should-have-b) hope will be helpful still as in the same content

**Comment by sherrif10** [comment]
@Ndacyayisenga-droid  some comments above to address .

**Pull request closed without merging**